crates/astria-composer/src/ds/eth_provider.rs

```diff
@@ -0,0 +1,92 @@
+use std::time::Duration;
+
+use async_trait::async_trait;
+use humantime::format_duration;
+use tracing::{debug, info, instrument, warn};
+use color_eyre::eyre::{self, Context};
+use tokio::sync::mpsc as tokio_mpsc;
+
+use ethers::providers::{
+    Middleware, Provider as EthersProvider, ProviderError as EthersProviderError, StreamExt, Ws,
+};
+
+use super::{streaming_client::StreamingClient, RollupChainId, RollupTxExt};
+
+/// A thin wrapper around [`Provider<Ws>`] to add timeouts.
+pub(crate) struct EthProvider {
+    inner: EthersProvider<Ws>,
+}
+
+// TODO(https://github.com/astriaorg/astria/issues/216): add timeouts for
+// `subscribe_full_pendings_txs` (more complex because it's a stream).
+impl EthProvider {
```

Comment:
this name is confusing since there's an `EthersProvider` and an `EthProvider`

Reply:
LOL, I'll try to do better but here Ethers means the library and Eth is the blockchain